source/draw/RasterYuv420Semiplanar.ooc

```diff
@@ -24,9 +23,36 @@ import io/FileReader
 import io/Reader
 import io/FileWriter
 
-RasterYuv420Semiplanar: class extends RasterYuvSemiplanar {
+RasterYuv420Semiplanar: class extends RasterImage {
+	_y: RasterMonochrome
+	_uv: RasterUv
+	y ::= this _y
+	uv ::= this _uv
 	stride ::= this _y stride
-	init: func ~fromRasterImages (yImage: RasterMonochrome, uvImage: RasterUv) { super(yImage, uvImage) }
+	crop: IntShell2D {
+		get
+		set (value) {
+			this crop = value
+			if (this y != null && this uv != null) {
+				this y crop = value
+				this uv crop = value / 2
+			}
+		}
+	}
+	init: func ~fromRasterImages (yImage: RasterMonochrome, uvImage: RasterUv) {
+		super(yImage size)
+		this _y = yImage
+		this _y referenceCount increase()
+		this _uv = uvImage
+		this _uv referenceCount increase()
+	}
+	init: func ~fromYuvSemiplanar (original: This, y: RasterMonochrome, uv: RasterUv) {
+		super(original)
+		this _y = y
+		this _y referenceCount increase()
+		this _uv = uv
+		this _uv referenceCount increase()
+	}
```

Comment:
These 16 constructors in different layers of inheritance are very redundant but I can't find a way to get rid of them.